### ISSUE TITLE
[Bug] Ghost variables to evaluate for loops are accessible to the user

### ISSUE THREAD

**Issue by aalok-t** [aptos-labs/aptos-core#11426]
# 🐛 Bug

Parsing the for loop creates a variable `__update_iter_flag` that is accessible to the user without them declaring it. This causes unexpected behavior such as follows.

## To reproduce

Consider the loop: 

`for (i in 0..10) { __update_iter_flag = false }`

This should not be valid as `__update_iter_flag` is not declared by the user. As a consequence, this access allows the user to create a non-terminating loop. 

## Expected Behavior

This should instead create an error that `__update_iter_flag` is not declared.

**Comment by brmataptos** [comment]
No, `__update_iter_flag` should be renamed to something not accepted as a user symbol, such as `$update_iter_flag`.

**Comment by brmataptos** [comment]
vineeth might be interested.   maybe standardizing the artificial sym names would be a good idea.

**Comment by brmataptos** [comment]
I priorized this as High because this is V1 and should be fixed before any release of this code, lest users start relying on it.
